MonoGame.Framework.Content.Pipeline/Graphics/GraphicsUtil.cs

```diff
@@ -17,29 +18,80 @@ internal static BitmapContent Resize(this BitmapContent bitmap, int newWidth, in
             BitmapContent src = bitmap;
             SurfaceFormat format;
             src.TryGetFormat(out format);
-            if (format != SurfaceFormat.Vector4)
+            var intermediateFormat = SurfaceFormat.Vector4;
+
+            switch (format)
+            {
+                case SurfaceFormat.Color or SurfaceFormat.Rgba64 or SurfaceFormat.Vector4:
+                    intermediateFormat = format;
+                    break;
+                default:
+                    var v4 = new PixelBitmapContent<Vector4>(src.Width, src.Height);
+                    BitmapContent.Copy(src, v4);
+                    src = v4;
+                    break;
+            }
+
+            var newBytes = new byte[intermediateFormat.GetSize() * newWidth * newHeight];
+
+            GCHandle srcHandle = default;
+            try
+            {
+                byte[] srcPixelData = src.GetPixelData();
+                srcHandle = GCHandle.Alloc(srcPixelData, GCHandleType.Pinned);
+                IntPtr srcPtr = srcHandle.AddrOfPinnedObject();
+
+                var srcBitmap = new MGCP_Bitmap
+                {
+                    width = src.Width,
+                    height = src.Height,
+                    type = TextureType.RgbaF,
+                    data = srcPtr,
+                };
+
+                var dstBitmap = new MGCP_Bitmap
+                {
+                    width = newWidth,
+                    height = newHeight,
+                };
+
+                IntPtr err = MGCP.MP_ResizeBitmap(ref srcBitmap, ref dstBitmap);
+                if (err != IntPtr.Zero)
+                {
+                    string errorMsg = System.Runtime.InteropServices.Marshal.PtrToStringUTF8(err);
+                    throw new InvalidContentException($"Bitmap resize failed: {errorMsg}");
+                }
+
+                Marshal.Copy(dstBitmap.data, newBytes, 0, newBytes.Length);
```

Comment:
@harry-cpp I wonder if we should have the C# side responsible for the memory of both the source and dest bitmaps passed to `MGCP.MP_ResizeBitmap`.  Then we potentially don't need this extra copy here?